src/fetcher.ts

```diff
@@ -81,35 +80,42 @@ export class BaseFetcher {
   ): Promise<T> {
     const controller = new AbortController()
     const id = setTimeout(() => controller.abort(), this.timeout)
-    const response: Response = await fetch(url, {
-      ...options,
-      signal: controller.signal
-    })
 
-    clearTimeout(id)
-    if (!response?.ok) {
-      throw new ApiError(
-        `Request failed with status ${response?.status} and message: ${response?.statusText}`
-      )
-    }
+    try {
+      const response: Response = await fetch(url, {
+        ...options,
+        signal: controller.signal
+      })
+      clearTimeout(id)
+
+      if (!response.ok) {
+        throw ApiError.createError(
+          `Request failed with status ${response.status}: ${response.statusText}`,
+          response.status
+        )
+      }
 
-    return response?.json() as T
+      return response.json() as Promise<T>
+    } catch (error: any) {
+      clearTimeout(id)
+      if (error.name === 'AbortError') {
```

Comment:
Add an optional chaining here too, just in case

```suggestion
      if (error?.name === 'AbortError') {
```